Add runtimes installation to the installer and remove obsolete installer codes

SadPencil opened the pull request:
Changes of this PR:
- Update InnoDependencyInstaller to the latest version, as a submodule. Remove the obsolete .NET core detection codes (not needed in the latest InnoDependencyInstaller and causes compilation error if they are not removed).
- Update Inno Setup to the latest v6.5.4 to fulfill the requirement of InnoDependencyInstaller.
- Update `installer.twig` file to include .NET 4.8 and VC++ runtimes as dependencies.
- Remove several obsolete runtime dependencies for .NET 4.0 and XNA 4.0, along with outdated files such as `Ra2.lcf` and `wsocks32.dll`. These components, which were functional when the installer was initially introduced [in 2020](https://github.com/CnCNet/cncnet-yr-client-package/commit/d0197cfc99f8915361b34f5df14fcf3562390baf#diff-2191597fce0c1e2925620ab67bf97d6d3ab396ea38456b0c9e29615261065863), ceased to work following our various updates. Even without this PR, these dependencies are obsolete and have been untriggerable for a significant period.
- Remove registering `Blowfish.dll` file as well writing `RA2.INI` since we have introduced Phobos. Also remove obsolete lines for `gamemd.exe` `game.exe` etc since we have introduced Phobos and no longer rely on the original game executables
- Remove obsolete registry lines (HKLM64 HKCU64) in Inno Setup script since x64 is now claimed
- Introduce `sortfilesbyextension` flag to get a ~3 MB free-of-charge solid compression advantage
- Handle case-sensitive case: use `RA2MD.INI` for `gamemd.exe` keys and use `RA2MD.ini` for the client usage, matching the current status.

Windows 7 SP1 (without any updates)

<img width="800" height="600" alt="Windows 7 SP1 en pure-2025-10-31-21-08-10" src="https://github.com/user-attachments/assets/292f1ff2-b74c-4b02-a3a9-9adac3264a48" />

Apart from the outdated SSL certificate (I don't have the patience to patch Windows 7 SP1 from scratch) the screenshots imply this method should work.

<img width="800" height="600" alt="Windows 7 SP1 en pure-2025-10-31-21-09-03" src="https://github.com/user-attachments/assets/da251e22-725a-4c86-bc3c-66ede85007d7" />

<img width="800" height="600" alt="Windows 7 SP1 en pure-2025-10-31-21-26-24" src="https://github.com/user-attachments/assets/65d163ae-bbfc-42c8-8e30-ac4ac146bda1" />

Also tested on my Windows 11 machine (with all components installed) and the installer works normally without downloading the runtimes.

SadPencil committed:
Upgrade InnoDependencyInstaller as a submodule

SadPencil committed:
Add .NET 4.8 and VC++ 2015-2022 as dependencies

SadPencil commented on tools/build-installer/inno/installer.twig:
```diff
@@ -291,8 +291,14 @@ var
+  //Dependency_ForceX86 := True; // force 32-bit install of next dependencies
+  Dependency_AddVC2015To2022;
+  //Dependency_ForceX86 := False; // disable forced 32-bit install again
+  Dependency_AddVC2015To2022;
```
nah

SadPencil committed:
Add submodule support to checkout action

SadPencil committed:
Remove obsolete .net core check

SadPencil committed:
Merge branch 'develop' into feature/update-inno-setup

SadPencil committed:
Upgrade Inno Setup to v6.5.4

SadPencil committed:
Reimplement GetFolderSize() in response to the submodule update

SadPencil committed:
Remove VC++ 2015-2022 dependency

SadPencil committed:
Add all VC++ runtimes

SadPencil committed:
oops

SadPencil committed:
Renormalize file

SadPencil committed:
Merge commit '40e0131' into update-inno-setup

SadPencil committed:
Merge branch 'develop' into update-inno-setup

Metadorius commented on tools/build-installer/inno/installer.twig:
```diff
@@ -286,13 +285,39 @@ begin
+  Dependency_AddVC2010;
+  Dependency_AddVC2012;
+  Dependency_AddVC2013;  
+  Dependency_AddVC2015To2022;
```
will this work correctly on x86 systems?

SadPencil replied:
yes since when `Dependency_ForceX86 = False` this is just the basic demo code provided by the library, automatically determines x86 or x64

Metadorius commented on tools/build-installer/inno/installer.twig:
```diff
@@ -286,13 +285,39 @@ begin
+  // Add dependencies here
+
+  // .NET Framework 4.8
+  Dependency_AddDotNet48;
```
will this correctly handle Windows 10 *not* having to install 4.8 because 4.8.1 is installed? and will this install 4.8.1 if for some reason Windows 10 installation doesn't have it?

SadPencil replied:
> will this correctly handle Windows 10 _not_ having to install 4.8 because 4.8.1 is installed? and will this install 4.8.1 if for some reason Windows 10 installation doesn't have it?

the first question: yes. it skipped installing 4.8 when 4.8.1 is installed, on my machine

the second question: I don't know. needs some test

Metadorius replied:
I am pretty sure if the first scenario is handled then the second is taken care of too

SadPencil replied:
> I am pretty sure if the first scenario is handled then the second is taken care of too

I doubt. The first question refers to a `reg_key_val <= const_number` determination which does not indicate the 2nd one be handled

SadPencil replied:
But it is hard to test this case though as the Windows 10 versions (ignoring end-of-life ones) comes with at least .net 4.8

Metadorius replied:
my point is that the people likely thought of that too

SadPencil replied:
Checked the source code. Answer to the 2nd question: no. It will acquire `ndp48-web.exe` file.

Metadorius replied:
On the second thought, this is alright, the chance of a win10 install without 481 is very low, and there's no easy way to tell whether it's not some version where 48 is supported but 481 isn't

Metadorius commented on tools/build-installer/class/template-model.class.ts:
```diff
@@ -8,6 +8,5 @@ export class TemplateModel {
     installDeleteFiles: string[];
-    excludedInstallerFiles: string;
-    netCoreCheckPath: string;
+    excludedInstallerFiles: string
```
is that supposed to not have `;`?

SadPencil replied:
optional when there is a line break. Added it back to maintain consistency

SadPencil committed:
Add the missing semi colon

SadPencil committed:
Uncomment ArchitecturesInstallIn64BitMode

SadPencil committed:
Remove the obsolete .NET 4.0 line

SadPencil committed:
Remove XNA redist call too since idpAddFile() has already been deleted

SadPencil committed:
Remove obsolete wic_x86_enu call too

since `ExtractTemporaryFile('wic_x86_enu.exe');` has been removed for a long time. This exe acts as a prerequisite for .NET 4.0

SadPencil committed:
Remove obsolete files

SadPencil committed:
Remove obsolete blowfish register

SadPencil committed:
Claim x64compatible

Co-Authored-By: Kerbiter <17500545+Metadorius@users.noreply.github.com>

SadPencil committed:
Remove unnecessary ra2.ini lines since Phobos is introduced to support campaign

SadPencil committed:
Remove redundant registry lines since we have claimed ArchitecturesInstallIn64BitMode=x64compatible

SadPencil committed:
Explicitly use `RA2MD.INI` in case for a case-sensitive file system in Wine

SadPencil committed:
Slightly change the comment of VC++ Redistributables (x64)

SadPencil committed:
Remove the 16-bit color compatibility setting

SadPencil committed:
Revert "Reimplement GetFolderSize() in response to the submodule update"

This reverts commit 016bb4acd325ee524394e89fb8145b4330c6c144.

SadPencil committed:
Merge branch 'develop' into update-inno-setup

SadPencil committed:
Add sortfilesbyextension to get 3 MB compression for free

SadPencil committed:
Use RA2MD.ini for client usages while RA2MD.INI for gamemd usages for now

SadPencil committed:
Claim original game exes as HIGHDPIAWARE, instead of removing the compatibility settings

SadPencil committed:
Hmmm... better don't touch them at all

SadPencil committed:
No need to touch game.exe and gamemd.exe for Wine anymore
